chrome/background.browserify.js

```diff
@@ -76,7 +76,8 @@ function onMessage(request, sender, sendResponse) {
           ) {
             // do not send login data to page if URL changed during search.
             if (tabs[0].url == request.urlDuringSearch) {
-              fillLoginForm(response, tabs[0]);
+              var autoSubmit = response.asOverride ? response.autoSubmit : getSettings().autoSubmit;
```

Comment:
why not simply check if `response.autoSubmit` is not undefined instead of introducing `asOverride`?

Reply:
Because go doesn't leave it undefined; it defaults to `false`.

Without `asOverride` there is no way to distinguish between a default `false`, and an explicitly specified `false` that should override the global setting.

Reply:
Let's try other approaches first before using `asOverride`, because it feels like a terrible hack...

First, you could try to make `autoSubmit` of a type `*bool`, a pointer to bool. That way the pointer can be `nil` (which is undefined) or `true/false`. I'm not sure how that would work with serialization, would it be transformed to `undefined` or not.

Another idea is to make `autoSubmit` a string, where you can use empty string as an indication of "value is not set".

Also, you can omit empty values completely during serialization, mark them as:

```
AutoSubmit string   `json:"autoSubmit,omitempty"`
```

Reply:
I had no idea `omitEMpty` was a thing - that's **much** nicer! I'll do that now, stand by...

Reply:
I'm not sure it will work for `bool` type though, but should work for `*bool`...

Reply:
How's that?

Reply:
With this approach, can you enable the auto-submit globally and override it with `autoSubmit: false` for one website?

I'm just worried that with `bool` type the `omitempty` considers `false` as empty, regardless of whether `false` is the default value or was set by you.

With `*bool` type, I would expect `omitempty` to understand that `nil` value is the empty value, while `true` and `false` are values that should not be omitted.

Reply:
`*bool` fixed it :-).

Reply:
That was a great idea of yours by the way - thank you :+1:.

Reply:
Thanks to you for implementing 😉  I'll test both PRs in the evening